packages/ember-htmlbars/tests/helpers/each_test.js

```diff
@@ -1,617 +0,0 @@
-
-import { registerKeyword, resetKeyword } from 'ember-htmlbars/tests/utils';
-import viewKeyword from 'ember-htmlbars/keywords/view';
-
-import buildOwner from 'container/tests/test-helpers/build-owner';
-import { OWNER } from 'container/owner';
-
-var people, view, owner;
-var template, templateMyView, MyView, MyEmptyView, templateMyEmptyView;
-var originalViewKeyword;
-
-var originalLookup = context.lookup;
-var lookup;
-
-import isEnabled from 'ember-metal/features';
-if (!isEnabled('ember-glimmer')) {
-  // jscs:disable
-
-QUnit.module('the #each helper', {
-  setup() {
-    context.lookup = lookup = {};
-
-    originalViewKeyword = registerKeyword('view',  viewKeyword);
-
-    template = compile('{{#each view.people as |person|}}{{person.name}}{{/each}}');
-    people = emberA([{ name: 'Steve Holt' }, { name: 'Annabelle' }]);
-
-    owner = buildOwner();
-
-    owner.register('view:toplevel', EmberView.extend());
-
-    view = EmberView.create({
-      [OWNER]: owner,
-      template: template,
-      people: people
-    });
-
-    templateMyView = compile('{{name}}');
-    lookup.MyView  = MyView = EmberView.extend({ template: templateMyView });
-    owner.register('view:my-view', MyView);
-
-    templateMyEmptyView = compile('I\'m empty');
-    lookup.MyEmptyView = MyEmptyView = EmberView.extend({
-      template: templateMyEmptyView
-    });
-    owner.register('view:my-empty-view', MyEmptyView);
-
-    runAppend(view);
-  },
-
-  teardown() {
-    runDestroy(owner);
-    runDestroy(view);
-    owner = view = null;
-
-    context.lookup = originalLookup;
-
-    resetKeyword('view', originalViewKeyword);
-  }
-});
-
-var assertHTML = function(view, expectedHTML) {
-  var html = view.$().html();
-
-  // IE 8 (and prior?) adds the \r\n
-  html = html.replace(/<script[^>]*><\/script>/ig, '').replace(/[\r\n]/g, '');
-
-  equal(html, expectedHTML);
-};
-
-QUnit.test('it renders the template for each item in an array', function() {
-  assertHTML(view, 'Steve HoltAnnabelle');
-});
-
-QUnit.test('it updates the view if an item is added', function() {
-  run(function() {
-    people.pushObject({ name: 'Tom Dale' });
-  });
-
-  assertHTML(view, 'Steve HoltAnnabelleTom Dale');
-});
-
-QUnit.test('it updates the view if an item is removed', function() {
-  run(function() {
-    people.removeAt(0);
-  });
-
-  assertHTML(view, 'Annabelle');
-});
-
-QUnit.test('it updates the view if an item is replaced', function() {
-  run(function() {
-    people.removeAt(0);
-    people.insertAt(0, { name: 'Kazuki' });
-  });
-
-  assertHTML(view, 'KazukiAnnabelle');
-});
-
-QUnit.test('can add and replace in the same runloop', function() {
-  run(function() {
-    people.pushObject({ name: 'Tom Dale' });
-    people.removeAt(0);
-    people.insertAt(0, { name: 'Kazuki' });
-  });
-
-  assertHTML(view, 'KazukiAnnabelleTom Dale');
-});
-
-QUnit.test('can add and replace the object before the add in the same runloop', function() {
```

Comment:
These covered cases where we were trying to track array changes instead of diff-ing, ages ago, even before htmlbars or glimmer, diff-ing had to be used to cover multiple array mutations in a run loop,  CollectionView had to switch to diff-ing childViews in order to make these tests pass.  The I-N-U-R array example is doing multiple mutations in one loop. So this is being covered